src/test/java/com/salesforce/dynamodbv2/UpdateTest.java

```diff
@@ -270,6 +494,180 @@ private void runUpdateConditionalOnGsiRkTest(TestArgument testArgument, boolean
         });
     }
 
+    @ParameterizedTest(name = "{arguments}")
+    @ArgumentsSource(DefaultArgumentProvider.class)
+    @DefaultArgumentProviderConfig(tables = { TABLE3 })
+    void updateWithAddConditionalOnGsiRkSuccess_usePlaceholder(TestArgument testArgument) {
+        runUpdateWithAddConditionalOnGsiRkTest(testArgument, true);
+    }
```

Comment:
Are these 4 tests really just validating that secondary index fields can't be updated with ADD? If so it'd be better to do a simpler test and merge these.